### PR TITLE
Make IntegrationTest use TimeTurner

### PR DESCRIPTION
**Pull request by reddvl1980**
Scaffolding 'plain' test for TimeTurner use

### CODE REVIEW

File: test/integration/IntegrationTest.cpp [hunk 1]
```diff
@@ -65,23 +65,46 @@ IntegrationTest::IntegrationTest()
...
 }
 
 void IntegrationTest::SetUp()
 {
+
+    if (__has_feature(address_sanitizer) || __has_feature(thread_sanitizer))
+    {
+        GTEST_SKIP();
+    }
+#if !ENABLE_LEGACY_API
```
Comment (olofkallander) [inline test/integration/IntegrationTest.cpp:80]: Is this something we do now only? I mean the tests contain both legacy and SMB API.

File: test/integration/IntegrationTest.cpp [hunk 2]
```diff
@@ -259,824 +292,843 @@ bool isActiveTalker(const std::vector<api::ConferenceEndpoint>& endpoints, const
...
+void IntegrationTest::finalizeSimulation()
+{
+    _internet->stop();
+    utils::Time::initialize();
+    _timeSource.shutdown();
+}
+
+TEST_F(IntegrationTest, plain)
+{
+    runTestInThread(_numWorkerThreads + 4, [this]() {
```
Comment (olofkallander) [inline test/integration/IntegrationTest.cpp:329]: We must look for a way around this.

**Review by olofkallander — APPROVED** [review]

File: test/integration/IntegrationTest.cpp (hunk 1 above)
Comment (reddvl1980) [reply test/integration/IntegrationTest.cpp:80]: This macros was present in all the tests, basically disabling tests if the build was not done with the _legaci API_ support. So this changes nothing. All tests would be run, but only on the build configs that:
a) does not use thread/address sanitizers
b) supports _legacy API_.

In future, when the _legacy API_ will be deprecated, we'll remove certain tests, refactor others and remove this check at all.

File: test/integration/IntegrationTest.cpp (hunk 2 above)
Comment (reddvl1980) [reply test/integration/IntegrationTest.cpp:329]: Please be more specific. Are you talking about number of threads we're waiting for or about lambda?

**Pull request merged**